app/models/post.rb

```diff
@@ -1,5 +1,48 @@
 class Post < ApplicationRecord
-  belongs_to :user
-  has_many :comments
-  has_many :likes
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  after_create :increment_user_counters
+  after_destroy :decrement_user_counters
+
+  after_create :increment_comments_counter
+  after_destroy :decrement_comments_counter
+
+  after_create :increment_likes_counter
+  after_destroy :decrement_likes_counter
+
+  def recent_comments(limit: 5)
+    comments.order(created_at: :desc).limit(limit)
+  end
+
+  private
+
+  def increment_user_counters
+    author.increment!(:posts_counter)
+  end
+
+  def decrement_user_counters
+    author.decrement!(:posts_counter)
+  end
+
+  def increment_comments_counter
+    update_columns(comments_counter: comments.count)
+  end
+
+  def decrement_comments_counter
+    return if destroyed?
+
+    update_columns(comments_counter: comments.count)
+  end
+
+  def increment_likes_counter
+    update_columns(likes_counter: likes.count)
+  end
+
+  def decrement_likes_counter
+    return if destroyed?
+
+    update_columns(likes_counter: likes.count)
+  end
```

Comment:
- [ ] [optional] @zunairkhan811 Although the previous reviewer didn't mention it the above methods are supposed to be implemented in their respective models according to the project description, kindly consider removing them as you have already implemented them in the said models.
![image](https://github.com/zunairkhan811/blog-app/assets/61408860/a6491946-6c0e-4b35-aa48-815f789ae40e)

Reply:
@danmainah Hi bro! The changes asked by you has been done. 😃  Here is the link to that.. [Click to see](https://github.com/zunairkhan811/blog-app/pull/3/commits/e735201156a46755bacada8bf4dc7b4b6f19af9a)